src/main/java/com/google/sps/data/PrioritizeCoursesCriteria.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.data;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+public class PrioritizeCoursesCriteria implements Preference {
+  List<Course> courseList;
+  
+  public PrioritizeCoursesCriteria(List<Course> courseList) {
+    this.courseList = courseList;
+  }
+  
+  /**
+   * Implements the preference score of the schedule, based on the criteria
+   * that prioritizes the order of courses that the user would like to take
+   */
+  public float preferenceScore(Schedule schedule) {
+    List<String> courseListNames = getCourseListNames(courseList);
+    int coursePrioritySum = 0;
+    List<Course> courses = (List) schedule.getCourses();
```

Comment:
You should almost always be suspicious of any typecasting. Here it works because the `Collection<Course>` that `schedule.getCoutses()` returns happens to be a `List<Collection>` but you have no guarantee of this. If you really need a `List<T>`, you can get one like this:

```
List<Course> courses = new ArrayList(schedule.getCourses());
```

For this bit of code though, you don't really need a `List<T>` because `Collection<Course>` supports the foreach syntax, which is what you need it for:

```
Collection<Course> courses = schedule.getCourses();

for (Course course : courses) {
  ...
}
```

Reply:
Thanks for point that out. It seems that I don't need the List<T>, so I changed it to Collection<T>